src/sentry/services/hybrid_cloud/auth/impl.py

```diff
@@ -130,6 +132,44 @@ def get_organization_key(self, *, key: str) -> Optional[RpcApiKey]:
         except ApiKey.DoesNotExist:
             return None
 
+    def enable_partner_sso(
+        self, *, organization_id: int, provider_key: str, provider_config: Mapping[str, Any]
+    ) -> None:
+        with enforce_constraints(transaction.atomic(router.db_for_write(AuthProvider))):
+            auth_provider = AuthProvider.objects.create(
+                organization_id=organization_id, provider=provider_key, config=provider_config
+            )
+
+            # TODO: Analytics requires a user id
+            # At provisioning time, no user is available so we cannot provide any user
+            # sso_enabled.send_robust(
+            #     organization=sentry_org,
+            #     provider=provider_key,
+            #     sender="EnablePartnerSSO",
+            # )
```

Comment:
Should the user_id be optional, and if it is present we trigger the signal?

Reply:
I've adding the `sso_enabled` signal creation in https://github.com/getsentry/getsentry/pull/11575.